src/config.rs

```diff
@@ -108,6 +114,103 @@ impl Config {
     pub fn demo_crates(&self) -> &DemoCrates {
         &self.demo_crates
     }
+
+    pub fn check() {
+        let result = Self::check_all();
+
+        if let Err(e) = result {
+            use std::process::exit;
+            error!("check-config failed: {}", e);
+            exit(1);
+        }
+    }
+
+    fn check_all() -> Result<()> {
+        // A simplified config definition, to validate it for duplicate keys.
+        #[derive(Deserialize)]
+        struct Config {
+            #[serde(rename = "crates")]
+            _crates: ::toml::Value,
+            #[serde(rename = "github-repos")]
+            _github_repos: ::toml::Value,
+        }
+
+        let buffer = Self::load_as_string()?;
+        let _cfg: Config = ::toml::from_str(&buffer)?;
+        let cfg: Self = ::toml::from_str(&buffer)?;
+        cfg.check_for_missing_crates()?;
+        cfg.check_for_missing_repos()?;
+        Ok(())
+    }
+
+    fn check_for_missing_crates(&self) -> Result<()> {
+        use crates::lists::{List, RegistryList};
+
+        if self.crates.is_empty() {
+            return Ok(());
+        }
+
+        info!("Fetching the crates index repo");
+        let reg = RegistryList;
+        let crates = reg.fetch()?;
```

Comment:
You should get the list from the database (`RegistryList.get(&db)`), asking the `&Database` from the `check` function and loading it from `cli.rs`. This way if someone already loaded the list they can use them, otherwise they can just run `create-lists`.